### PR TITLE
enable integration with AD for both ood and parallelcluster. Also add…

### PR DESCRIPTION
**Pull request by jxuamazon**
… example config files for pcluster

*Issue #, if available:*

*Description of changes:*
Added changes that are needed for ood and pcluster to both work with AD using AD users , instead of local users. 
Added example config files with two queues - one for simple job, one for ood interactive desktop apps. 
Added a post install script to install VNC, websocify fro interactive desktop. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cloudformation/openondemand.yml [hunk 1]
```diff
@@ -696,7 +696,8 @@ Resources:
             configure_cert:
               command: !Sub /etc/ood-install/open-on-demand-on-aws-${Branch}/scripts/configure_instance_cert.sh
           sources:
-            /etc/ood-install: !Sub https://github.com/aws-samples/open-on-demand-on-aws/archive/refs/heads/${Branch}.zip
+#            /etc/ood-install: !Sub https://github.com/aws-samples/open-on-demand-on-aws/archive/refs/heads/${Branch}.zip
+            /etc/ood-install: !Sub https://github.com/jxuamazon/open-on-demand-on-aws/archive/refs/heads/${Branch}.zip
```
Comment (cpollard0) [inline cloudformation/openondemand.yml:700]: Should we just parameterize this entire URL?

File: scripts/install_ood.sh [hunk 2]
```diff
@@ -82,20 +124,20 @@ echo "user_map_cmd: '/etc/ood/add_user.sh'" >> /etc/ood/config/ood_portal.yml
...
+#        useradd -M -u \$USERID \$USERNAME -d /shared/home/\$USERNAME
+#        usermod -a -G spack-users \$USERNAME
+#    fi
+#done < "/shared/userlistfile"
+#EOF
 
 chmod +x /etc/ood/add_user.sh
-chmod +x /shared/copy_users.sh
-chmod o+w /shared/userlistfile
+#chmod +x /shared/copy_users.sh
```
Comment (cpollard0) [inline scripts/install_ood.sh:139]: Let's just delete these instead of commenting htem out.

File: scripts/install_ood.sh [hunk 3]
```diff
@@ -11,6 +11,42 @@ export AD_SECRET=$(aws secretsmanager --region $AWS_REGION get-secret-value --se
...
+id_provider = ldap
+ldap_auth_disable_tls_never_use_in_production = true
+ldap_default_authtok = $AD_PASSWORD
+ldap_default_bind_dn = CN=Admin,OU=Users,OU=$DOMAIN_NAME,DC=$DOMAIN_NAME,DC=$TOP_LEVEL_DOMAIN
+ldap_id_mapping = True
+ldap_referrals = False
+ldap_schema = AD
+ldap_search_base = DC=$DOMAIN_NAME,DC=$TOP_LEVEL_DOMAIN
+ldap_tls_reqcert = never
+ldap_uri = ldap://$LDAP_NLB
```
Comment (cpollard0) [inline scripts/install_ood.sh:29]: Thanks for updating! I was testing and ran into this issue.

File: scripts/pcluster_head_node.sh [hunk 4]
```diff
@@ -107,27 +114,27 @@ chmod 400 /etc/munge/munge.key
...
-# ExecReload=/bin/kill -HUP $MAINPID
-# LimitNOFILE=65536
-# TasksMax=infinity
-# ExecStartPost=/bin/systemctl restart slurmctld
-
-# [Install]
-# WantedBy=multi-user.target
-
-# EOF
+cat <<EOF >> /etc/systemd/system/slurmdbd.service
```
Comment (cpollard0) [inline scripts/pcluster_head_node.sh:117]: Let's revert this change; it breaks PCluster SlurmAccounting integration for PCluster > 3.3.0 (at least)

File: scripts/create_sample_pcluster_config.sh [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            s3://$BUCKET_NAME/pcluster_worker_node_desktop.sh
+          Args:
+            - $STACK_NAME
+      Iam:
+        AdditionalIamPolicies:
+          - Policy: arn:aws:iam::aws:policy/AmazonS3FullAccess
+          - Policy: >-
+              $COMPUTE_POLICY
+  SlurmSettings: {}
+Region: us-east-1
```
Comment (cpollard0) [inline scripts/create_sample_pcluster_config.sh:114]: should this be $REGION instead of hardcoded to us-east-1?

File: scripts/create_sample_pcluster_config.sh [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+            Size: 50
+      CustomActions:
+        OnNodeConfigured:
+          Script: >-
+            s3://$BUCKET_NAME/pcluster_worker_node_desktop.sh
+          Args:
+            - $STACK_NAME
+      Iam:
+        AdditionalIamPolicies:
+          - Policy: arn:aws:iam::aws:policy/AmazonS3FullAccess
```
Comment (cpollard0) [inline scripts/create_sample_pcluster_config.sh:110]: Can you help me understand the need for AmazonS3FullAccess? I'm concerned it doesn't align with principal of least privilege.

Comment (jxuamazon) [reply scripts/create_sample_pcluster_config.sh]: removed.
